pkg/operator/controller/status.go

```diff
@@ -141,31 +162,52 @@ func computeOperatorStatusConditions(conditions []configv1.ClusterOperatorStatus
 	}
 	if numIngresses == ingressesAvailable {
 		progressingCondition.Status = configv1.ConditionFalse
+		progressingCondition.Message = ingressesEqualConditionMessage
 	} else {
 		progressingCondition.Status = configv1.ConditionTrue
 		progressingCondition.Reason = "Reconciling"
 		progressingCondition.Message = fmt.Sprintf(
 			"%d ingress controllers available, want %d",
 			ingressesAvailable, numIngresses)
 	}
-	conditions = setOperatorStatusCondition(conditions, progressingCondition)
+	if !strings.EqualFold(configVersion, currentVersion) {
+		progressingCondition.Message = fmt.Sprintf(
+			"configurd release version %q doesn't match current version %q",
```

Comment:
Typo: "configurd"→"configured".  I think saying "doesn't match current version" could be confusing, because once we update the clusteroperator status, the "current version" is now the configured version, right? So I think the message will be confusing in the context in which the administrator reads it.

Reply:
The intent is for @pravisankar to update https://github.com/openshift/cluster-ingress-operator/pull/200 to call `computeOperatorProgressingCondition()` after updating the operator status version, which would remove the `Progressing` condition message.

Another option is for this PR to remove the Progressing status message related to version inequality and have Ravi's PR set an upgrade-related message when computing the operator status version. Thoughts?

Reply:
> The intent is for @pravisankar to update #200 to call `computeOperatorProgressingCondition()` after updating the operator status version, which would remove the `Progressing` condition message in L175.

Would this all be happening in a single call to `syncOperatorStatus` (i.e., `syncOperatorStatus` would call `computeOperatorProgressingCondition`, then call `computeOperatorStatusVersions`, and then call `computeOperatorProgressingCondition` a second time), or would the first call to `syncOperatorStatus` set `Progressing` with the "doesn't match" message and a subsequent call to `syncOperatorStatus` update `.status.versions` and remove "doesn't match" message?

Are we only updating the message as a way to get `setOperatorStatusCondition` to update the condition's last transition, or will the message ever actually be updated in the API object?

Calling `computeOperatorProgressingCondition` twice in the same call to `syncOperatorStatus` seems non-obvious, and the result is that the message never actually gets published to the API, right?

> Another option is for this PR to remove the Progressing status message related to version inequality and have Ravi's PR set an upgrade-related message when computing the operator status version. Thoughts?

This seems closer to what I thought your intent was here: When we detect that the operator has been updated, immediately update the `Progressing` condition with a message (and, implicitly, a last transition time) that convey that an upgrade has taken place.

Reply:
> Would this all be happening in a single call to syncOperatorStatus (i.e., syncOperatorStatus would call computeOperatorProgressingCondition, then call computeOperatorStatusVersions, and then call computeOperatorProgressingCondition a second time), or would the first call to syncOperatorStatus set Progressing with the "doesn't match" message and a subsequent call to syncOperatorStatus update .status.versions and remove "doesn't match" message?

The call flow as you describe is correct. `computeOperatorStatusVersions` requires the status condition to be computed before computing the version. If `computeOperatorStatusVersions` makes any changes to status versions, it should then update `Progressing` to indicate "something" happened. I split `computeOperatorStatusConditions` into separate functions to provide a bit more efficiency.